src/context/auth-context.jsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+
+import { authReducer } from "../reducers";
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(authReducer, {
```

Comment:
Just a small suggestion, as we are going to have multiple states and dispatch so to differentiate, you can name it something as `[authState, authDispatch]`

Reply:
Done. Thanks for the suggestion!